src/ajax.js

```diff
@@ -333,7 +349,15 @@ jQuery.extend({
 			"text html": true,
 
 			// Evaluate text as a json expression
-			"text json": jQuery.parseJSON,
+			"text json": function(json) {
```

Comment:
Right, I was thinking that as well. We'd just add the ability for the converter to be async as you've done here.